mypy/newsemanal/semanal.py

```diff
@@ -1899,48 +1946,113 @@ def is_not_ready_type_ref(self, rv: Expression) -> bool:
                     return True
         return False
 
+    def should_wait(self, rv: Expression) -> bool:
```

Comment:
The name is pretty generic. What about renaming to `should_wait_rhs`, for example?